x/ibc/02-client/handler.go

```diff
@@ -10,43 +10,25 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
 	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/keeper"
 	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
-	localhosttypes "github.com/cosmos/cosmos-sdk/x/ibc/09-localhost/types"
 )
 
 // HandleMsgCreateClient defines the sdk.Handler for MsgCreateClient
-func HandleMsgCreateClient(ctx sdk.Context, k keeper.Keeper, msg exported.MsgCreateClient) (*sdk.Result, error) {
-	var (
-		consensusHeight uint64
-		clientState     exported.ClientState
-	)
-
-	switch msg.(type) {
-	// localhost is a special case that must initialize client state
-	// from context and not from msg
-	case *localhosttypes.MsgCreateClient:
-		clientState = localhosttypes.NewClientState(ctx.ChainID(), ctx.BlockHeight())
-		// Localhost consensus height is chain's blockheight
-		consensusHeight = uint64(ctx.BlockHeight())
```

Comment:
@colin-axner do you have any thoughts on the localhost client modularity? It is currently generated from the context

Reply:
Honestly, I think we should just only allow localhost to be created upon genesis. There is only one ever in existence (it doesn't really make sense to have multiple) and if a chain decides to init genesis without a localhost, anyone could just run `localhost.MsgCreateClient` and then it is created. I know removing this message would prevent localhost creation after genesis, but I think a localhost client should just be decided upon by the chain, not any individual. A chain could also make a module that just calls `CreateClient` so I don't think it'd be a huge issue.

I'm open to other opinions here. This is just my initial thoughts based on the current implementation. cc/ @AdityaSripal

Reply:
I agree with creating it only during genesis initialization. I think the most compelling use case for the localhost client is for testing so it doesn't make much sense to create one on the run.

Reply:
I agree with this